src/sentry/api/serializers/models/group.py

```diff
@@ -1081,20 +1082,20 @@ def serialize(self, obj, attrs, user):
         return result
 
     def _build_session_cache_key(self, project_id):
-        session_count_key = f"w-s:{project_id}"
-
+        start_key = end_key = env_key = ""
         if self.start:
-            session_count_key = f"{session_count_key}-{self.start.replace(minute=0, second=0, microsecond=0, tzinfo=None)}".replace(
-                " ", ""
-            )
+            start_key = self.start.replace(second=0, microsecond=0, tzinfo=None)
 
         if self.end:
-            session_count_key = f"{session_count_key}-{self.end.replace(minute=0, second=0, microsecond=0, tzinfo=None)}".replace(
-                " ", ""
-            )
+            end_key = self.end.replace(second=0, microsecond=0, tzinfo=None)
+
+        if self.end and self.start and self.end - self.start >= timedelta(minutes=60):
+            # Cache to the hour for longer time range queries, and to the minute if the query if for a time period under 1 hour
+            end_key = end_key.replace(minute=0)
+            start_key = start_key.replace(minute=0)
 
         if self.environment_ids:
-            envs = "-".join(str(eid) for eid in self.environment_ids)
-            session_count_key = f"{session_count_key}-{envs}"
+            env_key = "-".join(str(eid) for eid in self.environment_ids)
 
-        return session_count_key
+        session_cache_key = f"w-s:{project_id}-{start_key}-{end_key}-{env_key}".replace(" ", "")
```

Comment:
Not required or anything, but one option here is to just take all the key components and hash them, can be useful if the key is getting long (and saves you having to do the replace)